### PR TITLE
#1035 Changed the project lead and manager selects to be autocomplete

### LINKED ISSUE #1035: Activate Work Package - Change Project lead and manager selects to be autocomplete
### Description

Right now we have a select for project manager and lead, but theres too many values for a select on prod so we should make it an autocomplete instead

### Acceptance Criteria

Both are autocompletes

### Proposed Solution

Change the select to be an ner autocomplete in the activation change request file in the frontend. 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Aaryan1203**
Changed the project lead and manager selects to be NERAutocompletes
Screenshot 2023-04-05 at 8 23 31 PM
Screenshot 2023-04-05 at 8 56 08 PM
Screenshot 2023-04-05 at 8 57 45 PM

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1035

### CODE REVIEW

**Review by Peyton-McKee — APPROVED** [review]
AHHHHH THIS LOOKS SO GOOOD WWWWWWWW

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 1]
```diff
@@ -75,13 +74,17 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
+                    <NERAutocomplete
+                      id="project-lead-autocomplete"
+                      onChange={onChange}
+                      options={allUsers.map((p) => ({
+                        label: fullNamePipe(p),
+                        id: p.userId.toString()
+                      }))}
+                      size="small"
+                      placeholder="Project Lead"
+                      sx={{ width: 2 / 2 }}
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:86]: wait why do you need this

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 2]
```diff
@@ -94,13 +97,17 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
+                    <NERAutocomplete
+                      id="project-manager-autocomplete"
+                      onChange={onChange}
+                      options={allUsers.map((p) => ({
+                        label: fullNamePipe(p),
+                        id: p.userId.toString()
+                      }))}
+                      size="small"
+                      placeholder="Project Manager"
+                      sx={{ width: 2 / 2 }}
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:109]: also this why

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 3]
```diff
@@ -75,13 +74,17 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
-                    <Select onChange={onChange} value={value} variant="outlined" size="small" fullWidth>
-                      {allUsers.map((p) => (
-                        <MenuItem key={p.userId} value={p.userId}>
-                          {fullNamePipe(p)}
-                        </MenuItem>
-                      ))}
-                    </Select>
+                    <NERAutocomplete
+                      id="project-lead-autocomplete"
+                      onChange={onChange}
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:79]: oop also this doesnt work, you need to add the value={value} thing i think. Youll need to do the same thing for the other one.

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx (hunk 3 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:79]: oop Actually what you need to do is change the form input to take in the object { label: string, id: string } and then when you make the request extract the id from it cause we want the user id yes yes yes

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModalContainer.tsx [hunk 4]
```diff
@@ -20,8 +20,8 @@ interface ActivateWorkPackageModalContainerProps {
 }
 
 export interface FormInput {
-  projectLeadId: number | '';
-  projectManagerId: number | '';
+  projectLeadId?: number;
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModalContainer.tsx:23]: idk if you want me to just put this back, the fact that were using Form Input for default values maybe doesnt make much sense anymore? but i can also make the default values be like -1 if we dont want it to be optional?

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 5]
```diff
@@ -68,41 +82,29 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
-                    <Select onChange={onChange} value={value} variant="outlined" size="small" fullWidth>
-                      {allUsers.map((p) => (
-                        <MenuItem key={p.userId} value={p.userId}>
-                          {fullNamePipe(p)}
-                        </MenuItem>
-                      ))}
-                    </Select>
-                  </>
-                )}
+              <Typography>Project Lead</Typography>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:85]: don't need these titles, it's kinda redundant with the placeholder

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 6]
```diff
@@ -53,11 +50,28 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
+  const [projectManagerId, setProjectManagerId] = useState<string>();
+  const toast = useToast();
   /**
    * Wrapper function for onSubmit so that form data is reset after submit
    */
   const onSubmitWrapper = async (data: FormInput) => {
-    await onSubmit(data);
+    const { startDate, confirmDetails } = data;
+    if (!projectLeadId) {
+      toast.error('Please Set A Value For Project Lead');
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:62]: "Please Select a Project Lead"

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 7]
```diff
@@ -53,11 +50,28 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
+      return;
+    }
+    if (!projectManagerId) {
+      toast.error('Please Set A Value For Project Manager');
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:66]: same as above

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx [hunk 8]
```diff
@@ -68,41 +82,29 @@ const ActivateWorkPackageModal: React.FC<ActivateWorkPackageModalProps> = ({
...
-                    <Select onChange={onChange} value={value} variant="outlined" size="small" fullWidth>
-                      {allUsers.map((p) => (
-                        <MenuItem key={p.userId} value={p.userId}>
-                          {fullNamePipe(p)}
-                        </MenuItem>
-                      ))}
-                    </Select>
-                  </>
-                )}
+              <Typography>Project Manager</Typography>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModal.tsx:98]: The other two labels in this file should be form controls with form labels, can u update that while you're in here

File: src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModalContainer.tsx (hunk 4 above)
Comment (RChandler234) [reply src/frontend/src/pages/WorkPackageDetailPage/ActivateWorkPackageModalContainer/ActivateWorkPackageModalContainer.tsx:23]: ehhh it's fine for now, this whole solution for autocompletes is not my favorite, but tbh it works, and you made it look clean with the toasts, so I'm cool with it

**Review by RChandler234 — CHANGES_REQUESTED** [review]
image
when I go to scroll on the autocomplete it goes to scroll on the modal as well; should only have a scroll on the autocomplete dropdown

**Pull request merged**